apis/cloud.redhat.com/v1alpha1/namespacepool_types.go

```diff
@@ -38,6 +38,18 @@ type NamespacePoolSpec struct {
 	ResourceQuotas core.ResourceQuotaList `json:"resourcequotas"`
 	// Description for the namespace pool
 	Description string `json:"description,omitempty"`
+	// Contains a list of teams and corresponding secrets
+	Teams []Teams `json:"teams,omitempty"`
+}
+
+type Teams struct {
```

Comment:
Can we rename this to 'Team' ... because an array of `[]Team` would technically be Teams :)

```suggestion

// Team defines options that alter ENO behavior depending on name of team making the reservation
type Teams struct {
```